sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncClient.java

```diff
@@ -837,4 +989,765 @@ public Mono<Response<BinaryData>> getAppComponentsWithResponse(String testId, Re
+        bodyInBinaryData.getLength();
+        JsonMergePatchHelper.getTestAppComponentsAccessor().prepareModelForJsonMergePatch(body, false);
+        return createOrUpdateAppComponentsWithResponse(testId, bodyInBinaryData, requestOptions)
+            .flatMap(FluxUtil::toMono)
+            .map(protocolMethodData -> protocolMethodData.toObject(TestAppComponents.class));
+    }
+
+    /**
+     * Get associated app component (collection of azure resources) for the given test.
+     *
+     * @param testId Unique name for the load test, must contain only lower-case alphabetic,
+     * numeric, underscore or hyphen characters.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return associated app component (collection of azure resources) for the given test on successful completion of
+     * {@link Mono}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<TestAppComponents> getAppComponents(String testId) {
+        // Generated convenience method for getAppComponentsWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        return getAppComponentsWithResponse(testId, requestOptions).flatMap(FluxUtil::toMono)
+            .map(protocolMethodData -> protocolMethodData.toObject(TestAppComponents.class));
+    }
+
+    /**
+     * List server metrics configuration for the given test.
+     *
+     * @param testId Unique name for the load test, must contain only lower-case alphabetic,
+     * numeric, underscore or hyphen characters.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return test server metrics configuration on successful completion of {@link Mono}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<TestServerMetricsConfiguration> getServerMetricsConfig(String testId) {
```

Comment:
We should add unit tests with recordings for all these new APIs with models.

Reply:
Updated all the tests with new APIs models and recordings